### ISSUE TITLE
expo react native build failed after install @rnw-community/react-native-payments

### ISSUE THREAD

**Issue by majdhanna16** [rnw-community/rnw-community#203]
Hi,
After installing the package and running a build using eas build on cloud
the fastlane step start to failed 

Linking   Protein » Protein
⚠️  ld: ignoring duplicate libraries: '-lc++'
⚠️  ld: Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found
❌  Undefined symbols for architecture arm64
┌─ Symbol: _OBJC_CLASS_$_PKPaymentAuthorizationResult
└─ Referenced from: in react_native_payments[3](Payments.o)

❌  ld: symbol(s) not found for architecture arm64

❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)

⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » Protein/Protein » Build Phases » 'Start Packager'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » Protein/Protein » Build Phases » 'Bundle React Native code and images'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » Protein/Protein » Build Phases » '[CP-User] [RNFB] Core Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
    Run script build phase '[CP-User] Generate app.manifest for expo-updates' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based
▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	Ld /Users/expo/Library/Developer/Xcode/DerivedData/Protein-aaerlsxwczlppjhjrxvnoacxyptu/Build/Intermediates.noindex/ArchiveIntermediates/Protein/InstallationBuildProductsLocation/Applications/Protein.app/Protein normal (in target 'Protein' from project 'Protein')
▸ (1 failure)
** ARCHIVE FAILED **
The following build commands failed:
	Ld /Users/expo/Library/Developer/Xcode/DerivedData/Protein-aaerlsxwczlppjhjrxvnoacxyptu/Build/Intermediates.noindex/ArchiveIntermediates/Protein/InstallationBuildProductsLocation/Applications/Protein.app/Protein normal (in target 'Protein' from project 'Protein')
(1 failure)

Exit status: 65
+---------------------------------------+
|           Build environment           |
+-------------+-------------------------+
| xcode_path  | /Applications/Xcode.app |
| gym_version | 2.219.0                 |
| sdk         | iPhoneOS17.4.sdk        |
+-------------+-------------------------+
Looks like fastlane ran into a build/archive error with your project
It's hard to tell what's causing the error, so we wrote some guides on how
to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
Before submitting an issue on GitHub, please follow the guide above and make
sure your project is set up correctly.
fastlane uses `xcodebuild` commands to generate your binary, you can see the
the full commands printed out in yellow in the above log.
Make sure to inspect the output above, as usually you'll find more error information there
[!] Error building the application - see the log above
Error: The "Run fastlane" step failed with an unknown error. Refer to "Xcode Logs" below for additional, more detailed logs.

who can help with this issue?

**Comment by vitalyiegorov** [comment]
This package does not have a support for the `expo modules` I am planning to add this in the near future.

**Comment by NuskySainudeen** [comment]
Hi There, I am using a third party paymnet provider in my Expo reactnative app, there I need to process the Apple and Goolge pay. For that I used your @rnw-community/react-native-payments. After, implementing all the steps i cannot run the app for both Android and Ios. 

Can you halp me with this.

**Comment by jpetera54** [comment]
Hello,
I am going to expand on this issue. After adding the Passkit the whole app goes to white screen. Even App.js is not rendered. I am also using expo. Here is the AppDelagate.h of our app. When I remove the library, the app renders and functions normally even with Passkit added.

`
#import <RCTAppDelegate.h>
#import <UIKit/UIKit.h>
#import <PassKit/PassKit.h>
#import <Expo/Expo.h>

@interface AppDelegate : EXAppDelegateWrapper

@end
`

**Comment by WilliamRayo** [comment]
> Hello, I am going to expand on this issue. After adding the Passkit the whole app goes to white screen. Even App.js is not rendered. I am also using expo. Here is the AppDelagate.h of our app. When I remove the library, the app renders and functions normally even with Passkit added.
> 
> ```
> #import <RCTAppDelegate.h>
> #import <UIKit/UIKit.h>
> #import <PassKit/PassKit.h>
> #import <Expo/Expo.h>
> 
> @interface AppDelegate : EXAppDelegateWrapper
> 
> @end
> ```

I have the same issue for Android 👀 After installation i see only "white screen". When i remove this package from my Expo app, the app is running completely fine...

Also good to mention I followed the steps which are written in guidelines .. 🤷‍♂️

**Comment by KrastanD** [comment]
I ran across the same issue and the root cause was that `@rnw-community/react-native-payments` depends on `@rnw-community/platform` which depends on `react-native@0.72.17`. If you are on a newer version of react-native, when you install the package, you will have mismatched versions of react-native. The solution is to add a resolution to your package.json that pins the react-native version to the same one as your main react-native dependency. 
```
  "resolutions": {
     ...
    "react-native": "0.74.5",
  },
```
Ideally though, `@rnw-community/platform` shouldn't depend on a specific react-native version but have a minimum like: 
```  
"peerDependencies": {
    "react-native": ">=0.70",
}
```

The reason I saw the white screen was because I use a custom font, and when it doesn't load, the app returns null like:

```
  const [loaded] = useFonts({
    'Poppins-Regular': Poppins_400Regular,
    'Poppins-Bold': Poppins_600SemiBold,
  });

  if (!loaded) {
    return null;
  }
```

I assume you all have something similar and was probably recommended by expo-font at some point. The mismatched react-native versions lead to mismatched metro versions which caused expo-font to never load [and return this error](https://github.com/nrwl/nx/issues/27243)

I will be making the following change to my font loading code to give me some breadcrumbs if this happens again in the future.
```
  const [loaded, error] = useFonts({
    'Poppins-Regular': Poppins_400Regular,
    'Poppins-Bold': Poppins_600SemiBold,
  });

  if (!loaded) {
    error && console.error('Font errored during loading: ', error);
    return null;
  }
```
